packages/alchemy/src/middleware/userOperationSimulator.ts

```diff
@@ -2,21 +2,22 @@ import {
   deepHexlify,
   resolveProperties,
   type ClientMiddlewareFn,
+  type EntryPointVersion,
   type UserOperationStruct,
 } from "@alchemy/aa-core";
 import type { ClientWithAlchemyMethods } from "../client/types";
 
-export const alchemyUserOperationSimulator: <
-  C extends ClientWithAlchemyMethods
->(
-  client: C
-) => ClientMiddlewareFn =
-  (client) =>
-  async (struct, { account }) => {
+export function alchemyUserOperationSimulator<
+  C extends ClientWithAlchemyMethods,
+  TEntryPointVersion extends EntryPointVersion
+>(client: C): ClientMiddlewareFn {
+  return async (struct, { account }) => {
     const uoSimResult = await client.request({
       method: "alchemy_simulateUserOperationAssetChanges",
       params: [
-        deepHexlify(await resolveProperties(struct)) as UserOperationStruct,
+        deepHexlify(
+          await resolveProperties(struct)
+        ) as UserOperationStruct<TEntryPointVersion>,
```

Comment:
doesn't `alchemy_simulateUserOpeartionAssetChanges` support a union type? if so we should be able to leave this as it was and not pass in the generic here

Reply:
if not, then we should instead of doing `TEntryPointVersion extends EntryPointVersion` do `extends GetEntryPointFromAccount`

Reply:
ah its not generic anymore. will remove